mentat/llm_api.py

```diff
@@ -72,6 +73,12 @@ def raise_if_in_test_environment():
         raise MentatError("OpenAI call attempted in non benchmark test environment!")
 
 
+@backoff.on_exception(
+    wait_gen=backoff.expo, exception=Timeout, max_tries=5, base=2, factor=2
+)
+@backoff.on_exception(
+    wait_gen=backoff.expo, exception=RateLimitError, max_tries=3, base=2, factor=10
+)
```

Comment:
iirc these have some options for logging? It'd be nice to know how often these are being encountered / realize it's slowing things down

Reply:
It might also be good to write something to the terminal client if these are encountered as well. If the user keeps hitting `RateLimitErrors` then mentat will hang for 3 minutes before giving up right? That seems not ideal. Maybe a fringe case.

Reply:
It should only hang for 10 + 20 + 40 seconds, so a bit more than one minute; I think it's good as is, since if it fails all 3 times it does print the error it encountered, but if you still want to I can add that.

Reply:
Oh I misunderstood. I thought 2, 20, 200. That seems fine-ish. There's sort of a tension here where the optimal behavior in user and benchmark seems different:
* For benchmark retrying a dozen times would make sense. I don't want my overnight benchmark to fail because OpenAI had a short outage
* for users retry once in ~2 seconds then report to user seems best to me

I think the current change is good enough and better than the status quo but could be better

Reply:
I believe OpenAI's rate limit is reset every minute, and every day. So couldn't we just retry once, after 60 seconds, on `RateLimitError`?

Reply:
Hmm, I think you've convinced me, I'll add sending a message.